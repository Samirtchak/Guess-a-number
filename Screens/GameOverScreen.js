import React from 'react';
import {View,Text,StyleSheet,Button,Image} from 'react-native'

function GameOverScreen(props){
    return(
        <View style={styles.container}>
            <Text>Game Over</Text>
          <View style={styles.imageContainer}>
            <Image style={styles.image} resizeMode="cover" source={require('../assets/original.png')}/>
          </View>
            <Text>Number of rounds: {props.roundCount}</Text>
            <Text>Number was: {props.numberWin}</Text>
            <Button title="Restart" onPress={props.restart}/>

        </View>
    )

}
const styles = StyleSheet.create({
    container: {
        flex:1,
        alignItems:'center',
        justifyContent:'center'

    },
    imageContainer: {
        width:'80%',
        height:300,
        borderRadius: 200,
        borderWidth: 3,
        borderColor: 'black',
        overflow: 'hidden',
        marginVertical: 30
    },

    image: {
        width:'100%',
        height:300,
    }

})
export default GameOverScreen;