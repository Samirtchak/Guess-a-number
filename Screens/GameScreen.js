import React, {useState,useRef,useEffect} from  'react';
import {Text,View,StyleSheet,Button,Alert,ScrollView} from 'react-native';
import Card from "../components/Card";


const  generateRandomBetween = (min,max,exclude) => {
    min = Math.ceil(min);
    max = Math.floor(max);
    const rndNum  = Math.floor(Math.random() * (max - min)) + min;

    if(rndNum === exclude) {
        return generateRandomBetween(min,max,exclude);
    }
    else{
        return rndNum;
    }
}



 function GameScreen(props) {
    const initialGuess = generateRandomBetween(1,100,props.userChoice);
    const [currentGuess, setCurrentGuess] = useState(initialGuess);
     const[pastGuess, setPastGuess] = useState([initialGuess]);

     const currentLow=useRef(1);
    const currentHigh=useRef(100);

    useEffect(() =>{
        if(currentGuess === props.userChoice){
            props.onGameOver(pastGuess.length);
        }
    })


     const guessNextNumber = direction => {
         if((direction == 'Low' && currentGuess < props.userChoice)  || (direction == 'High' && currentGuess > props.userChoice)) {
             Alert.alert('Be honest','stop trying to lie.Be fair');
             return;
         }
         else{
             if(direction == 'Low'){
                 currentHigh.current=currentGuess;
             }
             else{
                 currentLow.current=currentGuess + 1;
             }

         }
         const nextGuess = generateRandomBetween(currentLow.current,currentHigh.current,currentGuess);
         setCurrentGuess(nextGuess);
         setPastGuess(currPastGuest => [nextGuess,...currPastGuest]);

             }

        return (
            <View style={styles.screen}>
                <Text style={styles.textSelect}>Computer guess number</Text>
                <Text style={styles.textNumber}>{currentGuess}</Text>
            <Card style={styles.buttonContainer}>
                <Button title="LOWER" onPress={guessNextNumber.bind(this,'Low')}/>
                <Button title="GREATER" onPress={guessNextNumber.bind(this,'High')}/>
            </Card>
                <Text>Current Guesses:</Text>
                <ScrollView>
                    {pastGuess.map(guess =><View key={guess}><Text>{guess}</Text></View>)}

                </ScrollView>


            </View>

        );
 }

 const styles = StyleSheet.create({
     screen:{
         flex:1,
         padding:10,
         alignItems:'center'
     },
     buttonContainer: {
         flexDirection:'row',
         justifyContent:'space-around',
         alignItems:'center',
         marginVertical:10,
         width:300,
         maxWidth:'80%'
     },
     textNumber: {
         paddingTop:10,
         color:'red',
         fontWeight:'bold',
         fontSize:40
     },
     textSelect: {
         fontSize:20,
         fontWeight:'bold'
     }

 })

export default GameScreen;