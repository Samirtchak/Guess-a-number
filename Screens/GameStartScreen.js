import React,{useState} from 'react';
import {StyleSheet, View, Text, TextInput, Button,Keyboard,TouchableWithoutFeedback} from 'react-native';
import Card from "../components/Card";
import Colors from "../constants/color";
import Input from "../components/Input";

function GameStartScreen(props) {
        const [enteredValue, setEnteredValue] = useState('');
        const inputEnteredValue = inputText => {
            setEnteredValue(inputText.replace(/[^0-9]/g,''));
        };

    return (
    <TouchableWithoutFeedback onPress={() => {Keyboard.dismiss()}}>
      <View style={styles.screen}>
          <Text style={styles.startGameTitle}>Start a New Game!</Text>
          <Card style={styles.inputContainer}>
            <Text>Select a Number</Text>
            <Input
                style={styles.inputStyle}
                blurOnSubmit
                autoCapitalize="none"
                autoCorrect={false}
                keyboardType="numeric"
                maxLength={2}
                onChangeText={inputEnteredValue}
                value={enteredValue}
            />
            <View style={styles.buttonContainer}>
                <View style={styles.button}><Button title="Reset" color={Colors.accent}/></View>
               <View style={styles.button}><Button title="Start" color={Colors.primary}/></View>

            </View>
          </Card>
      </View>
    </TouchableWithoutFeedback>


    );


}

const styles = StyleSheet.create({
    screen: {
        flex:1,
        paddingTop:10,
        alignItems:'center'
    },
    startGameTitle: {
      fontWeight:'bold',
      fontSize:20,
      marginVertical:10
    },

    inputContainer: {
        width:300,
        maxWidth:'80%',
        alignItems: 'center'
    },
    buttonContainer: {
        width:'80%',
        flexDirection:'row',
        justifyContent:'space-between'
    },
    button: {
        width:100
    },
    inputStyle:{
        textAlign:'center',
        width:50
    }

})

export default GameStartScreen;