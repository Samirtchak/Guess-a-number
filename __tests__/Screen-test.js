import React from 'react';
import GameStartScreen from '../Screens/GameStartScreen';
import GameScreen from '../Screens/GameScreen';
import GameOverScreen from '../Screens/GameOverScreen';


import renderer from 'react-test-renderer';

describe('Screen tests', () => {

    test(' GameStartScreen render correctly', () => {
        const tree = renderer.create(<GameStartScreen/>).toJSON();
        expect(tree).toMatchSnapshot();
    }); 
    test(' GameScreen render correctly', () => {
        const tree = renderer.create(<GameScreen/>).toJSON();
        expect(tree).toMatchSnapshot();
    }); 
    test(' GameOverScreen render correctly', () => {
        const tree = renderer.create(<GameOverScreen/>).toJSON();
        expect(tree).toMatchSnapshot();
    }); 

})
