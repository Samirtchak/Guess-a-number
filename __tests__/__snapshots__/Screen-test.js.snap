// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Screen tests  GameOverScreen render correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text>
    Game Over
  </Text>
  <View
    style={
      Object {
        "borderColor": "black",
        "borderRadius": 200,
        "borderWidth": 3,
        "height": 300,
        "marginVertical": 30,
        "overflow": "hidden",
        "width": "80%",
      }
    }
  >
    <Image
      resizeMode="cover"
      source={
        Object {
          "testUri": "../../../assets/original.png",
        }
      }
      style={
        Object {
          "height": 300,
          "width": "100%",
        }
      }
    />
  </View>
  <Text>
    Number of rounds: 
  </Text>
  <Text>
    Number was: 
  </Text>
  <View
    accessibilityRole="button"
    accessibilityStates={Array []}
    accessible={true}
    focusable={false}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Restart
      </Text>
    </View>
  </View>
</View>
`;

exports[`Screen tests  GameScreen render correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "padding": 10,
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 20,
        "fontWeight": "bold",
      }
    }
  >
    Computer guess number
  </Text>
  <Text
    style={
      Object {
        "color": "red",
        "fontSize": 40,
        "fontWeight": "bold",
        "paddingTop": 10,
      }
    }
  >
    58
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderRadius": 10,
        "elevation": 8,
        "flexDirection": "row",
        "justifyContent": "space-around",
        "marginVertical": 10,
        "maxWidth": "80%",
        "padding": 20,
        "shadowColor": "black",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.26,
        "shadowRadius": 6,
        "width": 300,
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityStates={Array []}
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          LOWER
        </Text>
      </View>
    </View>
    <View
      accessibilityRole="button"
      accessibilityStates={Array []}
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          GREATER
        </Text>
      </View>
    </View>
  </View>
  <Text>
    Current Guesses:
  </Text>
  <RCTScrollView>
    <View>
      <View>
        <Text>
          58
        </Text>
      </View>
    </View>
  </RCTScrollView>
</View>
`;

exports[`Screen tests  GameStartScreen render correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "paddingTop": 10,
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 20,
        "fontWeight": "bold",
        "marginVertical": 10,
      }
    }
  >
    Start a New Game!
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderRadius": 10,
        "elevation": 8,
        "maxWidth": "80%",
        "padding": 20,
        "shadowColor": "black",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.26,
        "shadowRadius": 6,
        "width": 300,
      }
    }
  >
    <Text>
      Select a Number
    </Text>
    <TextInput
      allowFontScaling={true}
      autoCapitalize="none"
      autoCorrect={false}
      blurOnSubmit={true}
      keyboardType="numeric"
      maxLength={2}
      onChangeText={[Function]}
      rejectResponderTermination={true}
      style={
        Object {
          "borderBottomColor": "grey",
          "borderBottomWidth": 1,
          "height": 30,
          "marginVertical": 10,
          "textAlign": "center",
          "width": 50,
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "width": "80%",
        }
      }
    >
      <View
        style={
          Object {
            "width": 100,
          }
        }
      >
        <View
          accessibilityRole="button"
          accessibilityStates={Array []}
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Array [
                Object {},
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#007AFF",
                    "fontSize": 18,
                    "padding": 8,
                    "textAlign": "center",
                  },
                  Object {
                    "color": "firebrick",
                  },
                ]
              }
            >
              Reset
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "width": 100,
          }
        }
      >
        <View
          accessibilityRole="button"
          accessibilityStates={Array []}
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Array [
                Object {},
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#007AFF",
                    "fontSize": 18,
                    "padding": 8,
                    "textAlign": "center",
                  },
                  Object {
                    "color": "dodgerblue",
                  },
                ]
              }
            >
              Start
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
